pandas/core/arrays/base.py

```diff
@@ -2628,7 +2628,15 @@ def _groupby_op(
             if op.how not in ["any", "all"]:
                 # Fail early to avoid conversion to object
                 op._get_cython_function(op.kind, op.how, np.dtype(object), False)
-            npvalues = self.to_numpy(object, na_value=np.nan)
+
+            arr = self
+            if op.how == "sum":
+                # https://github.com/pandas-dev/pandas/issues/60229
+                # All NA should result in the empty string.
+                assert "skipna" in kwargs
```

Comment:
What is the need for assert here?

Reply:
We should always be adding skipna to kwargs by the time we get here. If that doesn't happen for some reason (e.g. a future refactor), it can help debugging as an assert failing indicates a clear violation of an assumption, key error perhaps not. If it did somehow end up in a release with user code raising, an assert also indicates to the user "this is clearly a bug in pandas".

Reply:
@WillAyd - friendly ping